src/utils.rs

```diff
@@ -2,7 +2,9 @@ use std::io::Write;
 
 use secp256k1::hashes::{sha256, Hash};
 
-pub type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+use crate::error::Error;
+
+pub type Result<T> = std::result::Result<T, Error>;
```

Comment:
Ok I don't think that's a good idea to make this kind of aliases

Reply:
I've read around a little, it seems to be a common and accepted practice, nevermind

Reply:
:+1: it was more common than I thought, looks like even `std::io::Result<T>` is an alias for `Result<T, std::io::Error>`: https://doc.rust-lang.org/std/io/type.Result.html

I'll resolve this